flink-tests/src/test/java/org/apache/flink/runtime/metrics/JobManagerMetricsITCase.java

```diff
@@ -0,0 +1,186 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.metrics;
+
+import org.apache.flink.client.program.MiniClusterClient;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.testutils.BlockerSync;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.MetricConfig;
+import org.apache.flink.metrics.reporter.AbstractReporter;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.PrintSinkFunction;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import org.apache.flink.test.util.MiniClusterResource;
+import org.apache.flink.test.util.MiniClusterResourceConfiguration;
+
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Integration tests for proper initialization of the job manager metrics.
+ */
+public class JobManagerMetricsITCase {
+
+	private static final String JOB_MANAGER_METRICS_PREFIX = "localhost.jobmanager.";
+	private static final String TASK_SLOTS_AVAILABLE = "taskSlotsAvailable";
+	private static final String TASK_SLOTS_TOTAL = "taskSlotsTotal";
+	private static final String NUM_REGISTERED_TASK_MANAGERS = "numRegisteredTaskManagers";
+	private static final String NUM_RUNNING_JOBS = "numRunningJobs";
+
+	private static final BlockerSync sync = new BlockerSync();
+
+	private MiniClusterClient clusterClient;
+	private JobGraph jobGraph;
+
+	@ClassRule
+	public static final MiniClusterResource MINI_CLUSTER_RESOURCE = new MiniClusterResource(
+		new MiniClusterResourceConfiguration.Builder()
+			.setConfiguration(getConfiguration())
+			.setNumberTaskManagers(1)
+			.setNumberSlotsPerTaskManager(1)
+			.build());
+
+	@Before
+	public void setUp() throws Exception {
+		Assume.assumeTrue(
+			"ClusterClient is not an instance of MiniClusterClient",
+			MINI_CLUSTER_RESOURCE.getClusterClient() instanceof MiniClusterClient);
+
+		clusterClient = (MiniClusterClient) MINI_CLUSTER_RESOURCE.getClusterClient();
+		clusterClient.setDetached(true);
+
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
```

Comment:
you can put this job ad an `execute()` call into a separate thread and wait in the main test thread. then you don't have to create a jobgraph or use the cluster client.